packages/connector-core/src/index.ts

```diff
@@ -0,0 +1,30 @@
+import { ZodType } from 'zod';
+
+import { ConnectorError, ConnectorErrorCodes, GetConnectorConfig } from './types';
+
+export * from './types';
+
+// eslint-disable-next-line @silverhand/fp/no-let
+let _getConfig: GetConnectorConfig = async () => {
+  throw new Error('Not implemented');
+};
+
+export const implementGetConfig = (getConfig: GetConnectorConfig) => {
+  // eslint-disable-next-line @silverhand/fp/no-mutation
+  _getConfig = getConfig;
+};
+
+export function validateConfig<T>(config: unknown, guard: ZodType): asserts config is T {
```

Comment:
is there any reason not to use the arrow function definition?

Reply:
Arrow function cannot "assert"